widgets/src/lib/InferenceWidget/widgets/TextToImageWidget/TextToImageWidget.svelte

```diff
@@ -0,0 +1,128 @@
+<script>
+	import type { WidgetProps } from "../../shared/types";
+
+	import { onMount } from "svelte";
+	import WidgetQuickInput from "../../shared/WidgetQuickInput/WidgetQuickInput.svelte";
+	import WidgetWrapper from "../../shared/WidgetWrapper/WidgetWrapper.svelte";
+	import {
+		getDemoInputs,
+		getResponse,
+		getSearchParams,
+		updateUrl,
+	} from "../../shared/helpers";
+
+	export let apiToken: WidgetProps["apiToken"];
+	export let apiUrl: WidgetProps["apiUrl"];
+	export let callApiOnMount: WidgetProps["callApiOnMount"];
+	export let model: WidgetProps["model"];
+	export let noTitle: WidgetProps["noTitle"];
+	export let shouldUpdateUrl: WidgetProps["shouldUpdateUrl"];
+
+	let computeTime = "";
+	let error: string = "";
+	let isLoading = false;
+	let modelLoading = {
+		isLoading: false,
+		estimatedTime: 0,
+	};
+	let output = "";
+	let outputJson = "";
+	let text = "";
+
+	onMount(() => {
+		const [textParam] = getSearchParams(["text"]);
+		if (textParam) {
+			text = textParam;
+			getOutput();
+		} else {
+			const [demoText] = getDemoInputs(model, ["text"]);
+			text = (demoText as string) ?? "";
+			if (text && callApiOnMount) {
+				getOutput();
+			}
+		}
+	});
+
+	async function getOutput(withModelLoading = false) {
+		const trimmedText = text.trim();
+
+		if (!trimmedText) {
+			error = "You need to input some text";
+			output = "";
+			return;
+		}
+
+		if (shouldUpdateUrl) {
+			updateUrl({ text: trimmedText });
+		}
+
+		const requestBody = { inputs: trimmedText };
+
+		isLoading = true;
+
+		const res = await getResponse(
+			apiUrl,
+			model.modelId,
+			requestBody,
+			apiToken,
+			parseOutput,
+			withModelLoading
+		);
+
+		isLoading = false;
+		// Reset values
+		computeTime = "";
+		error = "";
+		modelLoading = { isLoading: false, estimatedTime: 0 };
+		output = "";
+		outputJson = "";
+
+		if (res.status === "success") {
+			computeTime = res.computeTime;
+			output = res.output;
+			outputJson = res.outputJson;
+		} else if (res.status === "loading-model") {
+			modelLoading = {
+				isLoading: true,
+				estimatedTime: res.estimatedTime,
+			};
+			getOutput(true);
+		} else if (res.status === "error") {
+			error = res.error || `Error encountered on input "${trimmedText}"`;
+		}
+	}
+
+	function parseOutput(body: unknown): string {
+		if (body && typeof body === "object" && body instanceof Blob) {
+			return URL.createObjectURL(body);
+		}
+		throw new TypeError(
+			"Invalid output: output must be of type object & of instance Blob"
+		);
+	}
+</script>
+
+<WidgetWrapper
+	{apiUrl}
+	{computeTime}
+	{error}
+	{model}
+	{modelLoading}
+	{noTitle}
+	{outputJson}
+>
+	<svelte:fragment slot="top">
+		<form>
+			<WidgetQuickInput
+				bind:value={text}
+				{isLoading}
+				onClickSubmitBtn={() => getOutput()}
```

Comment:
@beurkinger if I change this line to `onClickSubmitBtn={getOutput}`, I get typescript error:
```js
Type '(withModelLoading?: boolean) => Promise<void>' is not assignable to type '(e: MouseEvent) => void'.
  Types of parameters 'withModelLoading' and 'e' are incompatible.
    Type 'MouseEvent' is not assignable to type 'boolean'.ts(2322)
```

Reply:
Oh, I had not seen getOutput could take an argument. My bad !